app/src/main/java/net/kwatts/powtools/MainActivity.java

```diff
@@ -367,12 +368,33 @@ public void onPropertyChanged(Observable observable, int i) {
         mOWDevice.characteristics.get(MockOnewheelCharacteristicSpeed).value.addOnPropertyChangedCallback(new Observable.OnPropertyChangedCallback() {
             @Override
             public void onPropertyChanged(Observable observable, int i) {
-                alertsController.handleSpeed(mProgressiveGauge,mOWDevice.characteristics.get(MockOnewheelCharacteristicSpeed).value.get());
+                String speedString = mOWDevice.characteristics.get(MockOnewheelCharacteristicSpeed).value.get();
+                alertsController.handleSpeed(speedString);
+                updateGaugeOnSpeedChange(mProgressiveGauge, speedString);
             }
         });
         getBluetoothUtil().init(MainActivity.this, mOWDevice);
     }
 
+    private void updateGaugeOnSpeedChange(ProgressiveGauge gauge, @NonNull String speedString) {
+        net.kwatts.powtools.util.SharedPreferences sharedPreferences = App.INSTANCE.getSharedPreferences();
+        float speed;
+        try {
+            speed = Float.parseFloat(speedString);
+        } catch (NumberFormatException e) {
+            Timber.e(e);
+            return;
+        }
+        float speedAlert = sharedPreferences.getSpeedAlert();
+        boolean isEnabled = sharedPreferences.getSpeedAlertEnabled();
+
+        if (isEnabled && speed >= speedAlert) {
```

Comment:
I like pulling the gauge out, but unfortunate duplication of logic. Maybe this if check can live in a shared util?